File: THE DESCENT/The descent.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

int main()
{
    while (1) {
        int heights[8];
        int max = 0;
        for (int i = 0; i < 8; i++) {
            int mountainH; // represents the height of one mountain, from 9 to 0.
            cin >> mountainH; cin.ignore();
            heights[i] = mountainH;
            if (heights[max] < heights[i]) max = i;
        }

        cout << to_string(max) << endl;
    }
}

File: THE DESCENT/The descent.py
import sys
import math

while True:
    highestIndex = 0
    highestHeight = 0
    
    for i in range(8):
        mountain_h = int(input())  # represents the height of one mountain.
        if mountain_h > highestHeight:
            highestHeight = mountain_h
            highestIndex = i

    # The index of the mountain to fire on.
    print(highestIndex)